solr/core/src/java/org/apache/solr/search/DocSetQuery.java

```diff
@@ -78,11 +78,9 @@ public DocSet createDocSet(SolrIndexSearcher searcher) throws IOException {
 
     @Override
     public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
-        //This should probably use the provided boost as scorer. However, that causes
-        // TestSolrQueryParser.testFilter to fail.
-        return new ConstantScoreWeight(this, 0) {
+        return new ConstantScoreWeight(this, boost) {
```

Comment:
When I attempted refactoring this like so, `TestSolrQueryParser.testFilter` was failing. Granted, I didn't make the changes in FilterQuery. I am not sure I completely follow why the test now passes after making the changes in FilterQuery. Please explain.

Reply:
The net change is no difference for that test -- it's FilterQuery -> DocSetQuery and thus if *either* are scored zero, then zero times whatever is zero.

Reply:
Ah, ok got it, thank you